### PR TITLE
test(services): expand services test coverage

### PR DESCRIPTION
**Pull request by nikoksr**
## Description

Took the time and added some more mocks and tests. Added mocks and tests for 4 services (`msteams`, `amazonses`, `discord`, `slack`).

Also added a new make command, namely `mock`. All mock does is wrap `go generate ./...`. Services from now on should include a line similar to [this](https://github.com/nikoksr/notify/blob/17d70774847c755d8691e7682c7524a503b4524d/service/msteams/ms_teams.go#L10). This unifys the way that mocks are being generated and allows for quick re-generation of said mocks. Over time, additionally to adding more tests in general, more of the "older" tests should adapt this behavior, too.

## Motivation and Context

Since `Notify` should ideally have as close to 100% test coverage as possible, this is a step that brings us closer to that goal.

## How Has This Been Tested?

Ran the tests locally multiple times.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [x] Other  Test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

cc @svaloumas

### CODE REVIEW

File: service/msteams/ms_teams_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	service.AddReceivers("https://outlook.office.com/webhook/...", "https://outlook.office.com/webhook/...")
+	assert.Equal(3, len(service.webHooks))
+
+	hooks := []string{"https://outlook.office.com/webhook/...", "https://outlook.office.com/webhook/..."}
+	service.webHooks = []string{}
+	service.AddReceivers(hooks...)
+
+	diff := cmp.Diff(service.webHooks, hooks)
+	assert.Equal("", diff)
```
Comment (svaloumas) [inline service/msteams/ms_teams_test.go:41]:
```suggestion
    assert.Equal(service.webHooks, hooks)
```
I think it's more clear that way.

File: service/msteams/ms_teams_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	assert := require.New(t)
+
+	service := New()
+	assert.NotNil(service)
+}
+
+func TestMSTeams_AddReceivers(t *testing.T) {
```
Comment (svaloumas) [inline service/msteams/ms_teams_test.go:22]: I like that the prefix points to the service name. If we're going to keep this convention in the test files, I will open another PR today, renaming the rest of the test functions accordingly, to be consistent.

File: service/amazonses/amazon_ses_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ses"
+	"github.com/aws/aws-sdk-go-v2/service/ses/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
```
Comment (svaloumas) [inline service/amazonses/amazon_ses_test.go:14]:
```suggestion
func TestAmazonSES_New(t *testing.T) {
```

File: Makefile [hunk 4]
```diff
@@ -26,6 +26,11 @@ cover:
 	go test -race -covermode=atomic -coverprofile=coverage.out ./...
 .PHONY: cover
 
+mock:
```
Comment (svaloumas) [inline Makefile:30]: That's great! :+1: Should we document how to use it somewhere? Maybe a new `development` section in README (or a separate doc file) to accommodate information about how to generate mocks, write and run the tests, etc.

File: service/msteams/ms_teams_test.go (hunk 2 above)
Comment (nikoksr) [reply service/msteams/ms_teams_test.go:22]: If you are interested and have the time to change that, that would of course be great!

File: Makefile (hunk 4 above)
Comment (nikoksr) [reply Makefile:30]: I agree. I don't want the landing readme to get too bloated. I feel like it's aleady pushing it. We could add it to [CONTRIBUTING.md](https://github.com/nikoksr/notify/blob/main/CONTRIBUTING.md) tho.

File: service/msteams/ms_teams_test.go (hunk 2 above)
Comment (svaloumas) [reply service/msteams/ms_teams_test.go:21]: Will do!

**Pull request merged**